packages/material-ui/src/useScrollTrigger/useScrollTrigger.js

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+
+function getScrollY(ref = window) {
+  if (ref && ref.pageYOffset !== undefined) {
+    return ref.pageYOffset;
+  }
+  if (ref && ref.scrollTop !== undefined) {
+    return ref.scrollTop;
+  }
+  return document
+    ? (document.documentElement || document.body.parentNode || document.body).scrollTop
+    : 0;
+}
+
+function defaultTrigger(event, value, props = {}) {
+  const { directional = true, threshold = 100 } = props;
+  const scrollY = getScrollY(event.currentTarget);
+  // eslint-disable-next-line no-nested-ternary
+  const trigger = directional
+    ? scrollY < value.current
+      ? false
+      : !!(scrollY > value.current && scrollY > threshold)
+    : scrollY > threshold;
+
+  value.current = scrollY;
+  return trigger;
+}
+
+export default function useScrollTrigger(props = {}) {
+  const { onTriggerEval = defaultTrigger, ...initialOptions } = props;
```

Comment:
Should this handler be made private?

Reply:
I prefer the ability to override the handler for other use cases, like getScrollX.  It also leaves open the possibility of using this for a generic event trigger with some minor changes.  Making it private would defeat this purpose right?

Reply:
My main motivation for making it private is that it forces somebody that wants to extend the behavior of the hook to explicit his use case in an issue. It's informational. It helps to build better API. Now, we can try to anticipate it. If the use case is about only supporting scrollX, is this the best API?

Reply:
Yeah, you would have a better idea of the impact with your experience.  I can definitely see the argument and it would make it simpler both now and down the road.  Maybe we go with a direct solution now and the generic solution belongs elsewhere, maybe react-event-trigger ? :)